File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-fx-5-4/FxBoundary.cpp
// FxBoundary.cpp 
#include "FxBoundary.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

FxBoundary::FxBoundary(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //inclusiveNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* inclusiveNode = xmlNode->FirstChildElement("inclusive");

   if(inclusiveNode){inclusiveIsNull_ = false;}
   else{inclusiveIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- inclusiveNode , address : " << inclusiveNode << std::endl;
   #endif
   if(inclusiveNode)
   {
      if(inclusiveNode->Attribute("href") || inclusiveNode->Attribute("id"))
      {
          if(inclusiveNode->Attribute("id"))
          {
             inclusiveIDRef_ = inclusiveNode->Attribute("id");
             inclusive_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(inclusiveNode));
             inclusive_->setID(inclusiveIDRef_);
             IDManager::instance().setID(inclusiveIDRef_,inclusive_);
          }
          else if(inclusiveNode->Attribute("href")) { inclusiveIDRef_ = inclusiveNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { inclusive_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(inclusiveNode));}
   }

   //exclusiveNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* exclusiveNode = xmlNode->FirstChildElement("exclusive");

   if(exclusiveNode){exclusiveIsNull_ = false;}
   else{exclusiveIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- exclusiveNode , address : " << exclusiveNode << std::endl;
   #endif
   if(exclusiveNode)
   {
      if(exclusiveNode->Attribute("href") || exclusiveNode->Attribute("id"))
      {
          if(exclusiveNode->Attribute("id"))
          {
             exclusiveIDRef_ = exclusiveNode->Attribute("id");
             exclusive_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(exclusiveNode));
             exclusive_->setID(exclusiveIDRef_);
             IDManager::instance().setID(exclusiveIDRef_,exclusive_);
          }
          else if(exclusiveNode->Attribute("href")) { exclusiveIDRef_ = exclusiveNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { exclusive_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(exclusiveNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeBoolean> FxBoundary::getInclusive()
{
   if(!this->inclusiveIsNull_){
        if(inclusiveIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(inclusiveIDRef_));
        }else{
             return this->inclusive_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
boost::shared_ptr<XsdTypeBoolean> FxBoundary::getExclusive()
{
   if(!this->exclusiveIsNull_){
        if(exclusiveIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(exclusiveIDRef_));
        }else{
             return this->exclusive_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-fx-5-4/FxBoundary.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class FxBoundary : ISerialized
    {
        public FxBoundary(XmlNode xmlNode)
        {
            XmlNode inclusiveNode = xmlNode.SelectSingleNode("inclusive");
            
            if (inclusiveNode != null)
            {
                if (inclusiveNode.Attributes["href"] != null || inclusiveNode.Attributes["id"] != null) 
                {
                    if (inclusiveNode.Attributes["id"] != null) 
                    {
                        inclusiveIDRef_ = inclusiveNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(inclusiveNode);
                        IDManager.SetID(inclusiveIDRef_, ob);
                    }
                    else if (inclusiveNode.Attributes["href"] != null)
                    {
                        inclusiveIDRef_ = inclusiveNode.Attributes["href"].Value;
                    }
                    else
                    {
                        inclusive_ = new XsdTypeBoolean(inclusiveNode);
                    }
                }
                else
                {
                    inclusive_ = new XsdTypeBoolean(inclusiveNode);
                }
            }
            
        
            XmlNode exclusiveNode = xmlNode.SelectSingleNode("exclusive");
            
            if (exclusiveNode != null)
            {
                if (exclusiveNode.Attributes["href"] != null || exclusiveNode.Attributes["id"] != null) 
                {
                    if (exclusiveNode.Attributes["id"] != null) 
                    {
                        exclusiveIDRef_ = exclusiveNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(exclusiveNode);
                        IDManager.SetID(exclusiveIDRef_, ob);
                    }
                    else if (exclusiveNode.Attributes["href"] != null)
                    {
                        exclusiveIDRef_ = exclusiveNode.Attributes["href"].Value;
                    }
                    else
                    {
                        exclusive_ = new XsdTypeBoolean(exclusiveNode);
                    }
                }
                else
                {
                    exclusive_ = new XsdTypeBoolean(exclusiveNode);
                }
            }
            
        
        }
        
    
        #region Inclusive_
        private XsdTypeBoolean inclusive_;
        public XsdTypeBoolean Inclusive_
        {
            get
            {
                if (this.inclusive_ != null)
                {
                    return this.inclusive_; 
                }
                else if (this.inclusiveIDRef_ != null)
                {
                    inclusive_ = IDManager.getID(inclusiveIDRef_) as XsdTypeBoolean;
                    return this.inclusive_; 
                }
                else
                {
                      return this.inclusive_; 
                }
            }
            set
            {
                if (this.inclusive_ != value)
                {
                    this.inclusive_ = value;
                }
            }
        }
        #endregion
        
        public string inclusiveIDRef_ { get; set; }
        #region Exclusive_
        private XsdTypeBoolean exclusive_;
        public XsdTypeBoolean Exclusive_
        {
            get
            {
                if (this.exclusive_ != null)
                {
                    return this.exclusive_; 
                }
                else if (this.exclusiveIDRef_ != null)
                {
                    exclusive_ = IDManager.getID(exclusiveIDRef_) as XsdTypeBoolean;
                    return this.exclusive_; 
                }
                else
                {
                      return this.exclusive_; 
                }
            }
            set
            {
                if (this.exclusive_ != value)
                {
                    this.exclusive_ = value;
                }
            }
        }
        #endregion
        
        public string exclusiveIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

